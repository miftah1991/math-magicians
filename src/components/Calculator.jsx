import React, { useState } from 'react';
// prettier-ignore
import {
  Container,
  Row,
  Col,
  Button,
} from 'react-bootstrap';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
  const Calculator = () => {
    const [result, setResult] = useState({
      total: null,
      operation: null,
      next: null,
    });
  const clickHandler = (e) => {
    setResult(calculate(result, e.target.textContent));
  };
    return (
    <>
      <Container>
        <Row className="m-3">
          <Col>
            <div dir="ltr" className="w-100 p-2 result">
              {result.total}
              {result.operation}
              {result.next}
            </div>
          </Col>
        </Row>
        <Row className="m-3">
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>AC</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>+/-</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>%</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4 btn-warning" onClick={clickHandler} >
            รท
            </Button>
          </Col>
        </Row>
        <Row className="m-2">
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>7</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>8</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>9</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4 btn-warning" onClick={clickHandler}>
              x
            </Button>
          </Col>
        </Row>
        <Row className="m-2">
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>4</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>5</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>6</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4 btn-warning" onClick={clickHandler}>
              -
            </Button>
          </Col>
        </Row>
        <Row className="m-2">
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>1</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>2</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>3</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4 btn-warning" onClick={clickHandler}>
              +
            </Button>
          </Col>
        </Row>
        <Row className="m-2">
          <Col xs={6}>
            <Button className="w-100 p-4" onClick={clickHandler}>0</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4" onClick={clickHandler}>.</Button>
          </Col>
          <Col>
            <Button className="w-100 p-4 btn-warning" onClick={clickHandler}>
              =
            </Button>
          </Col>
        </Row>
      </Container>
      </>
    );
    };


export default Calculator;