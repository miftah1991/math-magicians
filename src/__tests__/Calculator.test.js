import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('check calculator render correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
