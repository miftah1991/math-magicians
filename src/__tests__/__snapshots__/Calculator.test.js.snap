// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check calculator render correctly 1`] = `
<div
  className="container"
>
  <div
    className="m-3 row"
  >
    <div
      className="col"
    >
      <div
        className="w-100 p-2 result"
        dir="ltr"
      />
    </div>
  </div>
  <div
    className="m-3 row"
  >
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn-warning btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
  </div>
  <div
    className="m-2 row"
  >
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn-warning btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
  </div>
  <div
    className="m-2 row"
  >
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn-warning btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
  </div>
  <div
    className="m-2 row"
  >
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn-warning btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="m-2 row"
  >
    <div
      className="col-6"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="w-100 p-4 btn-warning btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
